cmd/decrypt_value.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/base64"
+	"github.com/farmersedgeinc/yaml-crypt/pkg/actions"
+	"github.com/farmersedgeinc/yaml-crypt/pkg/cache"
+	"github.com/farmersedgeinc/yaml-crypt/pkg/config"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+	"strings"
+)
+
+var decryptValueFlags struct {
+}
+
+var decryptValueCmd = &cobra.Command{
+	Use:                   "decrypt-value",
+	Short:                 "Read in an encrypted value from STDIN and print an decrypted representation to STDOUT",
+	Long:                  "Read in an encrypted value from STDIN and print an decrypted representation to STDOUT. By default it will read in and decrypt a single line.",
+	Args:                  cobra.NoArgs,
+	DisableFlagsInUseLine: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return DecryptValue(os.Stdin, os.Stdout)
+	},
+}
+
+func DecryptValue(stdin io.Reader, stdout io.Writer) error {
+	reader := bufio.NewReader(stdin)
+	encodedCiphertext, err := reader.ReadString('\n')
+	if err != nil {
+		return err
+	}
+	encodedCiphertext = strings.TrimSpace(encodedCiphertext)
+	ciphertext, err := base64.StdEncoding.DecodeString(encodedCiphertext)
+	if err != nil {
+		return err
+	}
+	var plaintext string
+	func() error {
+		config, err := config.LoadConfig(".")
+		if err != nil {
+			return err
+		}
+		cache, err := cache.Setup(config)
+		if err != nil {
+			return err
+		}
+		defer cache.Close()
+		plaintext, err = actions.DecryptCiphertext(ciphertext, &cache, &config.Provider)
+		return err
+	}()
+	if err != nil {
+		return err
+	}
+	io.WriteString(stdout, plaintext+"\n")
```

Comment:
In the case of a single line, without a newline, I think shell ergonomics dictates that you do exactly this, add the newline.

What about the case of multi-line values, especially those ending with a newline?  what about if you printed to stderr "\nEnd of decrypted value" so that it would be obvious that there was a trailing newline?

So that it looks like this
```
alice@computer ~ % cat encrypted-stuff | yaml-crypt decrypt-value
Roses are red
Doggos are good
I'd pet every one
If only I could

END OF DECRYPTED VALUE
alice@computer ~ %
```

That way, the stderr wouldn't get in the way of piping stdout to other programs, but it would be super obvious there's a trailing newline.